### PR TITLE
Move dev_to_tty to compat

### PR DESCRIPTION
**Pull request by mpreisler**
It's also used in process probe and it was causing linker errors.

```
FAILED: tests/API/SEAP/test_api_sexp_ID 
: && /usr/lib64/ccache/cc -pipe -std=c99 -W -Wall -Wnonnull -Wshadow -Wformat -Wundef -Wno-unused-parameter -Wmissing-prototypes -Wno-unknown-pragmas -D_GNU_SOURCE -D_POSIX_C_SOURCE=200112L -O2 -DNDEBUG  -rdynamic tests/API/SEAP/CMakeFiles/test_api_sexp_ID.dir/test_api_sexp_ID.c.o  -o tests/API/SEAP/test_api_sexp_ID  -Wl,-rpath,/home/mpreisle/d/openscap/build/src src/libopenscap.so.8.13.0 -lxml2 -lxslt -lexslt -lpcre -lcurl -lpthread -lbz2 -lgcrypt && :
src/libopenscap.so.8.13.0: undefined reference to `dev_to_tty'
collect2: error: ld returned 1 exit status
ninja: build stopped: subcommand failed.
```

### CODE REVIEW

File: src/OVAL/probes/unix/process58-devname.h [hunk 1]
```diff
@@ -1,12 +0,0 @@
-
```
Comment (jan-cerny) [inline src/OVAL/probes/unix/process58-devname.h]: src/OVAL/probes/unix/process58_probe.c and src/OVAL/probes/unix/process_probe.c use this file

File: compat/CMakeLists.txt [hunk 2]
```diff
@@ -13,6 +13,10 @@ if(NOT HAVE_GETOPT_H)
 	list(APPEND COMPAT_SOURCES "getopt.c")
 endif()
 
+if(NOT HAVE_DEV_TO_TTY)
+	list(APPEND COMPAT_SOURCES "dev_to_tty.c")
```
Comment (jan-cerny) [inline compat/CMakeLists.txt]: It probably shouldn't be added on Windows.

File: compat/dev_to_tty.c [hunk 3]
```diff
@@ -30,7 +30,7 @@
 #include <fcntl.h>
 #include <unistd.h>
```
Comment (jan-cerny) [inline compat/dev_to_tty.c]: MSVC has problems with this file (it doesn't know `unistd.h`, doesn't support `restrict` keyword, dosn't have `readlink`, etc.). But it is used only in Unix probes, so it should be enough to remove it  from Windows builds.

Comment (mpreisler) [reply compat/dev_to_tty.c]: fixed

**Comment by jan-cerny** [comment]
It still doesn't build on Ubuntu. https://travis-ci.org/OpenSCAP/openscap/jobs/404322544#L1896

**Comment by jan-cerny** [comment]
I have found that the Travis CI currently is not configured to install `libprocps-dev`, which provides the header file that provides `dev_to_tty`. That means that the `compat` replacement for this function is built on Travis. It builds sucessfully, but then it is not able to link libopenscap.so.

**Comment by jan-cerny** [comment]
@openscap-ci test this please

**Pull request merged**